### ISSUE TITLE
create pytorch dataset

### ISSUE THREAD

**Issue by ArunB25** [ArunB25/MachineLearning_Project#9]
In the next task you will create an image classification model, but first, you need to create a dataset that feeds entries to the model.

First, you need to create an image dataset where each image is correctly labeled. To do so, you can use (for example) torch.utils.data.Dataset, simply create a class that inherits from torch.utils.data.Dataset and implement the __len__ and __getitem__ methods.

Remember, how do you load batches of data samples in PyTorch? You can use the torch.utils.data.DataLoader class, which is a wrapper around a dataset. It takes care of shuffling the data and loading batches of data.

There are a few things to remember when you create the Dataset:

The dataset should contain the images and the labels. You have two datasets, one for the images and one for the labels, so you should know what category correspond to what image.
You have to assign a label to each category, for example "Home & Garden" is category 0, "Appliances" is category 1, etc. This will be your encoder, which can be a dictionary. Thus, the output you obtain from the model will contain a list of numbers, which correspond to the categories, but you need to translate them. That's way, in addition to creating the encoder, you should also create a decoder.
While not necessary, you can add image transformations to your dataset. For example, rotate the images or flip them horizontally. This is called data augmentation.
A useful step here is to test that your DataLoader is working by running something like this: print(next(iter(dataloader))), where dataloader is the DataLoader you created.
